### PR TITLE
a step-by-step install guide on KIND k8s

### PR DESCRIPTION
**Pull request by bitsf**

### CODE REVIEW

**Comment by holyhope** [comment]
Is this PR still needed?
#27 is merged

File: docs/install-with-kind-k8s.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+  - containerPort: 443
+    hostPort: 443
+    protocol: TCP
+- role: worker
+- role: worker
+- role: worker
+EOF
+```
+
+- install go  gofmt  golangci-lint  kube-apiserver  kubebuilder  kubectl  kustomize  pkger etcd
```
Comment (steven-zou) [inline docs/install-with-kind-k8s.md]: @bitsf 

Could you please list the related links of installation documents of such tools?

**Comment by steven-zou** [comment]
> Is this PR still needed?
> #27 is merged

@bitsf 

I think you can update the extra contents to the installation doc introduced in #27?
cc @holyhope

**Comment by bitsf** [comment]
> Is this PR still needed?
> #27 is merged

I'm not sure yet, I just find the origin installation doc is more common, but user may still not know how to do in some specific case, or how to merge them.
What do you suggest, is this doc useful ?

File: docs/install-with-kind-k8s.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+```bash
+kubectl create namespace harbor-operator-system
+make deploy
+```
+
+- install nginx-ingess with nodeport
+```bash
+helm install nginx stable/nginx-ingress
+
+kubectl patch deployments nginx-nginx-ingress-controller -p '{"spec":{"template":{"spec":{"containers":[{"name":"nginx-ingress-controller","ports":[{"containerPort":80,"hostPort":80},{"containerPort":443,"hostPort":443}]}],"nodeSelector":{"ingress-ready":"true"},"tolerations":[{"key":"node-role.kubernetes.io/master","operator":"Equal","effect":"NoSchedule"}]}}}}'
```
Comment (holyhope) [inline docs/install-with-kind-k8s.md:55]: I suggested you changes about that awful patch: https://github.com/goharbor/harbor-operator/pull/27/commits/9fa9bffe412c1df6883101eb1a0a969c43bd2027
Please reuse it.

Comment (bitsf) [reply docs/install-with-kind-k8s.md]: not work yet

File: docs/install-with-kind-k8s.md (hunk 1 above)
Comment (bitsf) [reply docs/install-with-kind-k8s.md]: `make dev-tools` not fully work yet, so add some workaround.
You can help check if this workaround can be add to the makefile too

**Comment by steven-zou** [comment]
@bitsf @holyhope 

Let's keep this installation guide for Kind K8s as a separate document so that the user can easily follow to have a try.

**Pull request merged**